google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkService.java

```diff
@@ -40,7 +44,46 @@ public static CloudSdkService getInstance() {
 
   public abstract void setSdkHomePath(String path);
 
-  public abstract Set<CloudSdkValidationResult> validateCloudSdk(@NotNull Path pathToCloudSdk);
+  protected abstract Set<CloudSdkValidationResult> validateCloudSdk(Path path);
+
+  /**
+   * Checks if the stored path contains a valid Cloud SDK.
+   */
+  public Set<CloudSdkValidationResult> validateCloudSdk() {
+    return validateCloudSdk(getSdkHomePath());
+  }
+
+  /**
+   * Checks if a given path is malformed.
+   *
+   * <p>Windows' implementation of Paths doesn't handle well converting strings with certain special
+   * characters to paths. This method should be called before {@code Paths.get(path)}.
```

Comment:
Changed the first line to "Checks if a given path is malformed or if it contains a valid Cloud SDK."

The big motivation for this method really was this Windows issue we found, and it's the only use case for it right now, so I'd prefer to keep the second paragraph.